src/xmipp/libraries/reconstruction/angular_continuous_assign2.cpp

```diff
@@ -481,9 +503,14 @@ void ProgAngularContinuousAssign2::processImage(const FileName &fnImg, const Fil
 				steps(4)=steps(5)=steps(6)=1.;
 			if (optimizeAngles)
 				steps(7)=steps(8)=steps(9)=1.;
-			if (optimizeDefocus)
-				steps(10)=steps(11)=steps(12)=1.;
-			currentDefocusU = currentDefocusV = currentAngle = std::numeric_limits<double>::quiet_NaN(); // initialize default values
+			if (optimizeDefocus) {
+				if (sameDefocus)
+					steps(10)=steps(12)=1.; 
+				else {
+					steps(10)=steps(11)=steps(12)=1.;
+					currentDefocusU = currentDefocusV = currentAngle = std::numeric_limits<double>::quiet_NaN(); // initialize default values
```

Comment:
Is initializing to NaN a good starting point? Shouldn't it be better to take them from the CTF?